platform/javascript/detect.py

```diff
@@ -71,8 +71,9 @@ def configure(env):
         # -Os reduces file size by around 5 MiB over -O3. -Oz only saves about
         # 100 KiB over -Os, which does not justify the negative impact on
         # run-time performance.
-        env.Append(CCFLAGS=["-Os"])
-        env.Append(LINKFLAGS=["-Os"])
+        if env["optimize"] != "none":
```

Comment:
Should also be added to `release_debug` branch. Might be worth refactoring into `if env["target"].startswith("release"):` like done for iphone.